### ISSUE TITLE
Fields are missing when I `pip show` my Python package

### ISSUE THREAD

**Issue by jakelevi1996** [pypa/packaging-problems#670]
### OS version

Windows 10

### Python version

Python 3.7.6

### Pip version

23.1.2

### Guide link

https://packaging.python.org/tutorials/packaging-projects

### Problem description

(see related Stack Overflow question I asked [here](https://stackoverflow.com/q/76248637/8477566))

I recently uploaded my first Python package to PyPI. The relevant parts of `pyproject.toml` are defined as follows (full file available [here](https://github.com/jakelevi1996/jutility/blob/main/pyproject.toml)):

```
[build-system]
requires = ["setuptools>=61.0"]
build-backend = "setuptools.build_meta"

[project]
name = "jutility"
version = "0.0.4"
license = {text = "MIT License"}
authors = [
    {name = "Jake Levi", email = "jakelevi@hotmail.co.uk"},
]

# ...

[project.urls]
homepage = "https://github.com/jakelevi1996/jutility"
```

After installing this package with the command `python3 -m pip install -U jutility`, I run the command `python3 -m pip show jutility`, and get the following output:

```
Name: jutility
Version: 0.0.4
Summary: Collection of Python utilities intended to be useful for machine learning research and experiments
Home-page: 
Author: 
Author-email: Jake Levi <jakelevi@hotmail.co.uk>
License: MIT License
Location: /usr/local/lib/python3.10/dist-packages
Requires: matplotlib, numpy, Pillow
Required-by: 
```

Notably, the `Home-page` and `Author` fields are empty in the output from `pip show`, although they seem to be defined in `pyproject.toml`.

How should I change `pyproject.toml` to make these fields display properly in the `pip show` output?

Version-wise, I built and uploaded these packages to PyPI on my Windows 10 PC with Python 3.7.6, but I also tried downloading and installing this package and displaying the `pip show` output from a Google Colab notebook with Python 3.10.11. The package works completely as expected in the Colab notebook, but I get the same `pip show` output with empty `Home-page` and `Author` fields. I'd just like to know what I need to change in order to get these fields to display properly.

**Comment by henryiii** [comment]
This was discussed in [PEP 621](https://peps.python.org/pep-0621/). Author and Author-Email do not scale to multiple authors, so the non-optional Author-Email field is used with the `name <email>` format (Author has always been optional). And Home-page can be specified by the the more general, newer ([PEP 345](https://peps.python.org/pep-0345/))Project-URL field, so it's also expected to be empty. The goal is to avoid multiple places to specify the same thing ("redundant and confusing", according to the PEP).

FYI, this also includes License. While the original plan (in 2001, [PEP 241](https://peps.python.org/pep-0241/)) was to have a short string here for the license, by 2003 ([PEP 314](https://peps.python.org/pep-0314/)) is it specified via Classifiers, which correctly support specific licenses and multiple licenses, and this field is only there to allow text describing deviations from some known license. So you should not fill this in, and use Trove classifiers, unless you are using a non-standard license.

The problem is not the filling of the fields, but with `pip show`. That really needs to be updated to handle the current format better. I'm guessing the main issue might be worries about breaking anyone parsing the output of `pip show`? That's a wild guess, though, it might just be no one has updated it yet.

`setup.cfg` is an old file format that is no longer needed. It's only for setuptools, as well, while the pyproject.toml configuration works with many backend - setuptools>62, hatchling, flit-core, pdm-backend, scikit-build-core, meson-python, maturin, enscons, whey, and trampolim, for example.

**Comment by jakelevi1996** [comment]
- Thanks for your reply @henryiii
- I guess for me, it sounds like the optimal situation would be that all of the fields in `pip show` can be configured from `pyproject.toml`, and that the [Packaging Python Projects](https://packaging.python.org/en/latest/tutorials/packaging-projects/) tutorial explains what needs to be added to `pyproject.toml` in order to configure each of those fields (as mentioned in the first post above, currently the tutorial doesn't explain how to configure the `Home-page` and `Author` fields)
- As someone who's new to packaging Python projects, it doesn't matter to me too much how everything "should" work, other than that everything should be reasonably convenient and well-documented
- Even if `pyproject.toml` technically "shouldn't" configure the `Home-page` and `Author` fields of `pip show`, as someone who's new to packaging Python projects, when I see that prominent packages such as `numpy` and `matplotlib` display those fields in `pip show`, I'll naturally want to figure out how to configure those fields in my own package
- If I could configure everything from `pyproject.toml` then I'd happily get rid of `setup.cfg` (which I didn't use in the first place, and only added because it could easily fill in `pip show` fields that I couldn't work out how to fill in from `pyproject.toml`)

**Comment by henryiii** [comment]
The solution is to change `pip show`, not to add back deprecated and duplicated fields. FYI, numpy is switching to a PEP621 backend, so it will be dropping those fields from `pip show` soon. And most people aren't looking at `pip show`, they are looking at PyPI (which is better updated for changes in the ecosystem).

FYI, try pip show on `wheel` (0.40+), `build` (0.10+), or `packaging` (23+), and you'll see the same blank fields. It's not a problem with the standards (which are specified in PEPs), it's a problem with `pip show` (which is a single package).

**Comment by henryiii** [comment]
PEP 345 is showing the METADATA format, which is sort of YAML like (technically, it's based on the email module IIRC?), which is not related to the TOML format. Your build backend (setuptools, hatching, flit-core, etc) will convert the standard TOML fields defined in PEP 621 into METADATA. There isn't a field for Home-Page or Author, and there's not really one for License either (the PEP 621 definition didn't match the METADATA definition, and is being revised in [PEP 639](https://peps.python.org/pep-0639/)).

`pip show` isn't _supposed_ to have backward compatibly issues, since you aren't supposed to parse the output, but rather use METADATA yourself if you want to process it, but yes, I'm sure someone does. 🤦
